addons/stock/models/stock_production_lot.py

```diff
@@ -37,6 +37,20 @@ def create(self, vals):
                 raise UserError(_("You are not allowed to create a lot for this picking type"))
         return super(ProductionLot, self).create(vals)
 
+    @api.multi
+    def write(self, vals):
+        if 'product_id' in vals:
+            move_lines = self.env['stock.move.line'].search([
+                ('lot_id', 'in', self.ids), ('move_id.state', '!=', 'cancel')
```

Comment:
If it has cancelled ones this should not be changed either.

IMHO, we should be able to delete the cancelled moves on hide them (active=False) instead, but this domain should not have the part `('move_id.state', '!=', 'cancel')`.

Reply:
@sle-odoo any opinion about this?

Reply:
Isn't better to check for stock.quant records with that lot?

Reply:
@pedrobaeza v11.0 (AFAICU) remove all relevance to quant, now they are **only** for cache purposes, now the control is over the lines and the moves.

Reply:
@nhomar i agree, even if cancelled move lines do not appear in the traceability report, if when you read it the lot_id is not linked anymore to the original product it becomes really hard to understand what happened
@pedrobaeza you cannot look at the quants because it is possible to track goods in your stock and stop the tracking at the delivery, so you'll get a quant in customer without lot_id
@nim-odoo i would remove the second par of the domain. thank you

Reply:
@sle-odoo then the quant doesn't update the lot when delivering to customer and it's specified in the delivery picking? :scream: